tool/dart_model_generator/lib/generate_dart_model.dart

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:dart_style/dart_style.dart';
+import 'package:json_schema/json_schema.dart';
+
+class DartModelGenerator {
```

Comment:
Does this need to be a class? It seems like really just a namespace. Consider at least moving all the methods to static methods, but I would probably move it all to top level methods.